### ISSUE TITLE
‘ENOBJ_ITER_myevent_alarm’ was not declared in this scope

### ISSUE THREAD

**Issue by faissaloo** [enigma-dev/enigma-dev#981]
Make 3 objects, A,B and C.  
Make A the child of B.  
Give B a collision event with C.  
Give A an alarm event.  
Result:  
```
In file included from SHELLmain.cpp:110:0:
/home/faissaloo/.enigma/Preprocessor_Environment_Editable/IDE_EDIT_objectdeclarations.h: In destructor ‘virtual enigma::OBJ_child::~OBJ_child()’:
/home/faissaloo/.enigma/Preprocessor_Environment_Editable/IDE_EDIT_objectdeclarations.h:334:14: error: ‘ENOBJ_ITER_myevent_alarm’ was not declared in this scope
       delete ENOBJ_ITER_myevent_alarm;
              ^
make[1]: Leaving directory '/home/faissaloo/enigma-dev/ENIGMAsystem/SHELL'
Makefile:12: recipe for target 'Game' failed
make: *** [Game] Error 2
```

**Comment by RobertBColton** [comment]
This is a duplicate of #870

This issue was not the original #870 but did creep up after Josh attempted to cleanup the inheritance code and missed an important line, or else it was always that way, I can't tell at this point.

But the fix is https://github.com/enigma-dev/enigma-dev/issues/870#issuecomment-226436467

But that does not completely resolve #870 because there are still the issues @sorlok mentioned.

**Comment by faissaloo** [comment]
Is there a particular reason why the fix hasn't been put into master yet?

**Comment by RobertBColton** [comment]
Well I have controller and input fixes in my local copy and I didn't have time or want to mess around checking out another ENIGMA to do it. Plus I had no one to verify that the changes didn't break any other games (from my own testing it doesn't). 

Also I wanted to try to clean up the code while also fixing that bug and do it all in one PR, but really it doesn't matter. One thing I wanted to change was where I check if any code = "" and replace it with code.empty() which is the more correct way to check if a string is empty (i.e. I do this to check if the parent event has no code).

If you want to apply the fix and send a PR we'll merge it if it doesn't break anything for you. On my end it fixes the very first example on the EDC, which is the "Platform Engine".
http://enigma-dev.org/edc/games.php?action=list

**Comment by faissaloo** [comment]
I'll give it a go after we deal with the rounded rectangle business (unless there's a way to have more than one fork of the same project that I don't know of on here).

**Comment by RobertBColton** [comment]
When you do please change anywhere I had a comparison of a string with = "" to use str.empty(). Other issues can be worked out as we find them but since I already know about these ones we should go ahead and include it.

**Comment by faissaloo** [comment]
Will do.

**Comment by RobertBColton** [comment]
I've gone ahead and created the fix in #987, please test it when you have time to verify it fixes the issue in your game.
